test/test.js

```diff
@@ -1,85 +1,55 @@
 'use strict';
-/* eslint-env mocha */
-var assert = require('assert');
-var execFile = require('child_process').execFile;
-var fs = require('fs');
-var path = require('path');
-var binCheck = require('bin-check');
-var BinBuild = require('bin-build');
-var compareSize = require('compare-size');
-var mkdirp = require('mkdirp');
-var rimraf = require('rimraf');
 
-var tmp = path.join(__dirname, 'tmp');
-
-beforeEach(function (cb) {
-	mkdirp(tmp, cb);
+const path = require('path');
+const cp = require('child_process');
+const test = require('ava');
+const pify = require('pify');
+const binCheck = require('bin-check');
+const BinBuild = require('bin-build');
+const compareSize = require('compare-size');
+const pathExists = require('path-exists');
+const mkdirp = pify(require('mkdirp'));
+const rimraf = pify(require('rimraf'));
+const pngquant = require('../');
+
+const execFile = pify(cp.execFile);
+const tmp = path.join(__dirname, 'tmp');
+
+test.beforeEach(async () => {
+	await mkdirp(tmp);
 });
 
-afterEach(function (cb) {
-	rimraf(tmp, cb);
+test.afterEach(async () => {
+	await rimraf(tmp);
 });
```

Comment:
Would be better to use an actual temp directory, so these `beforeEach` and `afterEach` hooks wouldn't be needed. You can use: https://github.com/sindresorhus/tempy